srml/democracy/src/lib.rs

```diff
@@ -431,47 +714,92 @@ impl<T: Trait> Module<T> {
 		Self::deposit_event(RawEvent::Executed(index, ok));
 	}
 
+	/// Table the next waiting proposal for a vote.
 	fn launch_next(now: T::BlockNumber) -> Result {
+		if <LastTabledWasExternal<T>>::take() {
+			Self::launch_public(now).or_else(|_| Self::launch_external(now))
+		} else {
+			Self::launch_external(now).or_else(|_| Self::launch_public(now))
+		}.map_err(|_| "No proposals waiting")
+	}
+
+	/// Table the waiting external proposal for a vote, if there is one.
+	fn launch_external(now: T::BlockNumber) -> Result {
+		if let Some((proposal, threshold)) = <NextExternal<T>>::take() {
+			<LastTabledWasExternal<T>>::put(true);
+			Self::deposit_event(RawEvent::ExternalTabled);
+			Self::inject_referendum(
+				now + T::VotingPeriod::get(),
+				proposal,
+				threshold,
+				T::EnactmentPeriod::get(),
+			)?;
+			Ok(())
+		} else {
+			Err("No external proposal waiting")
+		}
+	}
+
+	/// Table the waiting public proposal with the highest backing for a vote.
+	fn launch_public(now: T::BlockNumber) -> Result {
 		let mut public_props = Self::public_props();
 		if let Some((winner_index, _)) = public_props.iter()
 			.enumerate()
-			.max_by_key(|x| Self::locked_for((x.1).0).unwrap_or_else(Zero::zero)/*defensive only: All current public proposals have an amount locked*/)
+			.max_by_key(|x| Self::locked_for((x.1).0).unwrap_or_else(Zero::zero)
+				/* ^^ defensive only: All current public proposals have an amount locked*/)
```

Comment:
suggest moving the `)` that's at the end of this line to the end of the previous line instead